Sources/SwiftUICoordinator/Routing/NavigationController.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  NavigationController.swift
+//  
+//
+//  Created by Erik Drobne on 12/05/2023.
+//
+
+import Foundation
+import SwiftUI
+
+public class NavigationController: UINavigationController {
+    
+    // MARK: - Properties
+    
+    public private(set) var transitions = [Transition]()
+    
+    // MARK: - Initialization
+    
+    public override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        
+        self.delegate = self
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    public convenience init() {
+        self.init(nibName: nil, bundle: nil)
+    }
+    
+    // MARK: - Public methods
+    
+    public func register(_ transition: Transition) {
+        guard transitions.first(where: { element in
```

Comment:
could you use a dictionary here instead of an array? Might be less time complex

Reply:
Yes, will do that! Thanks!